// x-global-api/dist/esm/global-api.js
import './global-api.css.js';
import { Toast } from './toast/index.js';
import { Modal } from './modal/index.js';

let toastInstance = null;
let modalInstance = null;
const showToast = (options) => {
  if (!toastInstance) {
    toastInstance = new Toast();
  }
  toastInstance.show(options);
};
const hideToast = () => {
  console.log(toastInstance, "toastInstance");
  toastInstance?.hide();
};
const showLoading = (options) => {
  const newOptions = Object.assign(
    {
      iconUrl: "https://guild-1251316161.file.myqcloud.com/common/icon-loading-white.png",
      title: "\u52A0\u8F7D\u4E2D..."
    },
    { ...options },
    {
      duration: 0
    }
  );
  showToast(newOptions);
};
const hideLoading = () => {
  hideToast();
};
const showModal = (options) => {
  if (!modalInstance) {
    modalInstance = new Modal();
  }
  modalInstance.show(options);
};
const hideModal = () => {
  modalInstance?.hide();
};

export { hideLoading, hideModal, hideToast, showLoading, showModal, showToast };
