// x-global-api/dist/index.js
(function(e,g){typeof exports=="object"&&typeof module!="undefined"?g(exports):typeof define=="function"&&define.amd?define(["exports"],g):(e=typeof globalThis!="undefined"?globalThis:e||self,g(e.globalApi={}))})(this,function(e){"use strict";function g(i,n){n===void 0&&(n={});var o=n.insertAt;if(!(!i||typeof document=="undefined")){var a=document.head||document.getElementsByTagName("head")[0],t=document.createElement("style");t.type="text/css",o==="top"&&a.firstChild?a.insertBefore(t,a.firstChild):a.appendChild(t),t.styleSheet?t.styleSheet.cssText=i:t.appendChild(document.createTextNode(i))}}var T=`/* variables */
/* -------------------------------------- */
:root {
  --global-api-toast-bg-color: rgba(0, 0, 0, 0.7);
  --global-api-toast-text-color: #fff;
  --global-api-toast-radius: 8px;
  --global-api-toast-z-index: 6000;

  --global-api-modal-overlay-bg-color: rgba(0, 0, 0, 0.7);
  --global-api-modal-bg-color: #fff;
  --global-api-modal-bg-active-color: #ededed;
  --global-api-modal-text-color: #000;
  --global-api-modal-radius: 8px;
  --global-api-modal-z-index: 5000;
  --global-api-modal-btn-border-color: #ededed;
}

/* toast */
/* -------------------------------------- */
.global-api-toast-overlay {
  position: fixed;
  z-index: var(--global-api-toast-z-index);
  background: transparent;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.global-api-toast-main {
  position: fixed;
  z-index: var(--global-api-toast-z-index);
  background: var(--global-api-toast-bg-color);
  color: var(--global-api-toast-text-color);
  border-radius: var(--global-api-toast-radius);
  box-sizing: border-box;
  overflow: hidden;
  padding: 10px;
  left: 50%;
  transform: translate(-50%, -50%);
}

.global-api-toast-main--top,
.global-api-toast-main--bottom {
  transform: translate(-50%, 0);
}

.global-api-toast-main--inline {
  display: flex;
  align-items: center;
}

.global-api-toast-main--inline .global-api-toast-icon {
  width: 24px;
  height: 24px;
  margin: 0 5px 0 0;
  flex-shrink: 0;
}

.global-api-toast-icon {
  width: 48px;
  height: 48px;
  margin: 5px 10px;
  background-size: cover;
  background-repeat: no-repeat;
  margin-left: auto;
  margin-right: auto;
}

.global-api-toast-title {
  font-size: 14px;
  line-height: 26px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

/* modal */
/* -------------------------------------- */
.global-api-modal-overlay {
  position: fixed;
  z-index: var(--global-api-modal-z-index);
  background: var(--global-api-modal-overlay-bg-color);
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.global-api-modal-main {
  position: fixed;
  z-index: var(--global-api-modal-z-index);
  background: var(--global-api-modal-bg-color);
  color: var(--global-api-modal-text-color);
  border-radius: var(--global-api-modal-radius);
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  overflow: hidden;
  font-size: 16px;
}

.global-api-modal-title {
  padding-top: 25px;
  font-weight: bold;
  text-align: center;
  line-height: 1;
}

.global-api-modal-content {
  text-align: center;
  line-height: 1.8;
  padding: 30px 16px;
}

.global-api-modal-footer--inline {
  display: flex;
  position: relative;
}

.global-api-modal-footer-btn {
  flex: 1;
  text-align: center;
  line-height: 50px;
  position: relative;
  cursor: pointer;
  user-select: none;
  font-weight: bold;
}

.global-api-modal-footer-btn:active {
  background-color: var(--global-api-modal-bg-active-color);
}

.global-api-modal-footer--inline::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 1px;
  background: var(--global-api-modal-btn-border-color);
  transform: scaleY(0.5);
}

.global-api-modal-footer--inline .global-api-modal-footer-btn:not(:first-of-type)::before {
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  width: 1px;
  background: var(--global-api-modal-btn-border-color);
  transform: scaleX(0.5);
}

.global-api-modal-footer--block .global-api-modal-footer-btn::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 1px;
  background: var(--global-api-modal-btn-border-color);
  transform: scaleY(0.5);
}

/* animation: fade in/out */
/* -------------------------------------- */
.global-api-fade-in {
  animation: globalApiFadeIn 0.25s ease-in-out both;
}

.global-api-fade-out {
  animation: globalApiFadeOut 0.25s ease-in-out both;
}

@keyframes globalApiFadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes globalApiFadeOut {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}

/* animation: down in/out */
/* -------------------------------------- */
.global-api-down-in {
  animation: globalApiDownIn 0.25s ease-in-out both;
}

.global-api-down-out {
  animation: globalApiDownOut 0.25s ease-in-out both;
}

@keyframes globalApiDownIn {
  from {
    opacity: 0;
    transform: translate(-50%, calc(-100% - 30px));
  }
  to {
    opacity: 1;
    transform: translate(-50%, 0);
  }
}

@keyframes globalApiDownOut {
  from {
    opacity: 1;
    transform: translate(-50%, 0);
  }
  to {
    opacity: 0;
    transform: translate(-50%, calc(-100% - 30px));
  }
}

/* animation: up in/out */
/* -------------------------------------- */
.global-api-up-in {
  animation: globalApiUpIn 0.25s ease-in-out both;
}

.global-api-up-out {
  animation: globalApiUpOut 0.25s ease-in-out both;
}

@keyframes globalApiUpIn {
  from {
    opacity: 0;
    transform: translate(-50%, calc(100% + 30px));
  }
  to {
    opacity: 1;
    transform: translate(-50%, 0);
  }
}

@keyframes globalApiUpOut {
  from {
    opacity: 1;
    transform: translate(-50%, 0);
  }
  to {
    opacity: 0;
    transform: translate(-50%, calc(100% + 30px));
  }
}
`;g(T);class k{constructor(){this.parent=document.body,this.wrap=null,this.isHiding=!1,this.iconMap={success:"data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTYgMTYiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGNpcmNsZSBjeD0iOCIgY3k9IjgiIHI9IjgiIGZpbGw9IiMxNUQxNzMiPjwvY2lyY2xlPjxwYXRoIGQ9Ik00LjU1MTUxIDcuNjAwMUw3LjE4MjEzIDEwLjIzMDZMMTEuOTE2MSA1LjQ5NjY4IiBzdHJva2U9IndoaXRlIiBzdHJva2Utd2lkdGg9IjEuNSIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCI+PC9wYXRoPjwvc3ZnPg==",error:"data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTYgMTYiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGNpcmNsZSBjeD0iOCIgY3k9IjgiIHI9IjgiIGZpbGw9IiNGRjU3NjUiPjwvY2lyY2xlPjxwYXRoIGQ9Ik04IDEwVjEyIiBzdHJva2U9IndoaXRlIiBzdHJva2Utd2lkdGg9IjIiPjwvcGF0aD48cGF0aCBkPSJNOCA0VjkiIHN0cm9rZT0id2hpdGUiIHN0cm9rZS13aWR0aD0iMiI+PC9wYXRoPjwvc3ZnPg==",warning:"data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTYgMTYiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PGNpcmNsZSBjeD0iOCIgY3k9IjgiIHI9IjgiIGZpbGw9IiMwMDk5RkYiPjwvY2lyY2xlPjxwYXRoIGQ9Ik04IDdWMTIiIHN0cm9rZT0id2hpdGUiIHN0cm9rZS13aWR0aD0iMiI+PC9wYXRoPjxwYXRoIGQ9Ik04IDRWNiIgc3Ryb2tlPSJ3aGl0ZSIgc3Ryb2tlLXdpZHRoPSIyIj48L3BhdGg+PC9zdmc+Cg==",custom:"",none:""},this.initOptions()}initOptions(){this.options={icon:"none",mask:!1,duration:2e3,maxWidth:"auto",place:"center",offset:"10px",layout:"block",animation:"fade"}}show(n){if(this.wrap){this.hide(()=>{this.show(n)});return}const{animation:o,place:a}=n;if(!o&&a&&a!=="center"){const s={top:"down",bottom:"up"};n.animation=s[a]}Object.assign(this.options,n);const{title:t,iconUrl:l,parent:r,duration:d}=n;if(l&&(this.iconMap.custom=l,this.options.icon="custom"),this.options.icon==="none"&&!t){console.error("icon or title is required!");return}if(r&&(this.parent=r),this.generateHTML(),d){const s=setTimeout(()=>{this.hide(),clearTimeout(s)},d)}}hide(n){if(!this.wrap||this.isHiding)return;this.isHiding=!0;const{animation:o}=this.options,a=this.wrap.querySelector(".global-api-toast-main"),t=()=>{a==null||a.removeEventListener("animationend",t),this.parent.removeChild(this.wrap),this.wrap=null,this.isHiding=!1,this.initOptions(),n==null||n()};a==null||a.addEventListener("animationend",t),a==null||a.classList.add(`global-api-${o}-out`)}generateHTML(){const{place:n,offset:o,layout:a,mask:t,icon:l,maxWidth:r,title:d,wrapClass:s}=this.options,{animation:u}=this.options;this.wrap=document.createElement("div"),this.wrap.classList.add("global-api-toast"),s&&this.wrap.classList.add(s);let c=`global-api-toast-main global-api-${u}-in`;a==="inline"&&(c+=" global-api-toast-main--inline");let m="top: 50%; ";console.log("place",n),n==="top"&&(m=`top: ${o};`,c+=" global-api-toast-main--top"),n==="bottom"&&(m=`bottom: ${o};`,c+=" global-api-toast-main--bottom");const f=t?'<div class="global-api-toast-overlay"></div>':"",v=`<div class="${c}" style="max-width: ${r}; ${m}">`,w=l==="none"?"":`<div class="global-api-toast-icon" style="background-image: url(${this.iconMap[l]})"></div>`,y=d?`<div class="global-api-toast-title">${d}</div>`:"",I="</div>",x=f+v+w+y+I;this.wrap.innerHTML=x,this.parent.appendChild(this.wrap)}}class L{constructor(){this.parent=document.body,this.wrap=null,this.footer=null,this.overlay=null,this.isHiding=!1,this.initOptions()}initOptions(){this.options={width:"300px",footerTexts:[{text:"\u53D6\u6D88"},{text:"\u786E\u8BA4",color:"#00cafc"}],footerLayout:"inline",maskCanClose:!0,animation:"fade"}}show(n){if(this.wrap){this.hide(()=>{this.show(n)});return}Object.assign(this.options,n);const{title:o,content:a,parent:t}=n;if(!a&&!o){console.error("content or title is required!");return}t&&(this.parent=t),this.generateHTML(),this.onEventListener()}generateHTML(){const{wrapClass:n,footerTexts:o,footerLayout:a,width:t,title:l,content:r,animation:d}=this.options;this.wrap=document.createElement("div"),this.wrap.classList.add("global-api-modal"),n&&this.wrap.classList.add(n);const s=`global-api-modal-main global-api-${d}-in`,u=`global-api-modal-footer global-api-modal-footer--${a}`,c=o.reduce((C,h,S)=>{const Y=h.color?`color: ${h.color};`:"";return C+`<div class="global-api-modal-footer-btn" data-key="${h.key||S}" style="${Y}">${h.text}</div>`},""),m='<div class="global-api-modal-overlay global-api-fade-in"></div>',f=`<div class="${s}" style="width: ${t};">`,v=l?`<div class="global-api-modal-title">${l}</div>`:"",w=r?`<div class="global-api-modal-content">${r}</div>`:"",y=`<div class="${u}">${c}</div>`,I="</div>",x=m+f+v+w+y+I;this.wrap.innerHTML=x,this.parent.appendChild(this.wrap)}onEventListener(){var n,o;this.footer=this.wrap.querySelector(".global-api-modal-footer"),this.overlay=this.wrap.querySelector(".global-api-overlay"),(n=this.footer)==null||n.addEventListener("click",this.footerHandler.bind(this)),this.options.maskCanClose&&((o=this.overlay)==null||o.addEventListener("click",this.maskHandler.bind(this)))}offEventListener(){var n,o;(n=this.footer)==null||n.removeEventListener("click",this.footerHandler.bind(this)),(o=this.overlay)==null||o.removeEventListener("click",this.maskHandler.bind(this))}footerHandler(n){const{className:o,dataset:a}=n.target,{callback:t}=this.options;o.includes("global-api-modal-footer-btn")&&(t==null?void 0:t(a.key))!==!0&&this.hide()}maskHandler(){this.hide()}hide(n){if(!this.wrap||this.isHiding)return;this.isHiding=!0;const{animation:o}=this.options,a=this.wrap.querySelector(".global-api-modal-main"),t=this.wrap.querySelector(".global-api-modal-overlay"),l=()=>{a==null||a.removeEventListener("animationend",l),this.offEventListener(),this.parent.removeChild(this.wrap),this.wrap=null,this.isHiding=!1,this.initOptions(),n==null||n()};a==null||a.addEventListener("animationend",l),a==null||a.classList.add(`global-api-${o}-out`),t==null||t.classList.add("global-api-fade-out")}}var G=Object.defineProperty,j=Object.getOwnPropertySymbols,z=Object.prototype.hasOwnProperty,O=Object.prototype.propertyIsEnumerable,H=(i,n,o)=>n in i?G(i,n,{enumerable:!0,configurable:!0,writable:!0,value:o}):i[n]=o,N=(i,n)=>{for(var o in n||(n={}))z.call(n,o)&&H(i,o,n[o]);if(j)for(var o of j(n))O.call(n,o)&&H(i,o,n[o]);return i};let p=null,b=null;const M=i=>{p||(p=new k),p.show(i)},P=()=>{console.log(p,"toastInstance"),p==null||p.hide()},Z=i=>{const n=Object.assign({iconUrl:"https://guild-1251316161.file.myqcloud.com/common/icon-loading-white.png",title:"\u52A0\u8F7D\u4E2D..."},N({},i),{duration:0});M(n)},D=()=>{P()},E=i=>{b||(b=new L),b.show(i)},A=()=>{b==null||b.hide()};e.Modal=L,e.Toast=k,e.hideLoading=D,e.hideModal=A,e.hideToast=P,e.showLoading=Z,e.showModal=E,e.showToast=M});
